include/xsimd/arch/xsimd_avx.hpp

```diff
@@ -680,6 +684,25 @@ namespace xsimd
             return _mm256_add_pd(tmp1, tmp2);
         }
 
+        // insert
+        template <class A, class T, size_t I, class = typename std::enable_if<std::is_integral<T>::value, void>::type>
+        inline batch<T, A> insert(batch<T, A> const& self, T val, index<I> pos, requires_arch<avx>) noexcept
+        {
+            switch (sizeof(T))
+            {
+#ifndef _WIN32
```

Comment:
Why doesn't this apply on Windows? Did you intend `_MSC_VER` @serge-sans-paille ?

Reply:
> Why doesn't this apply on Windows? Did you intend `_MSC_VER` @serge-sans-paille ?

This was failing using the windows compiler, and also with clang-cl due to invalid intrinsics, if I recall correctly. Feel free to propose a patch and investigate :-)